api/src/auth/drivers/oauth2.ts

```diff
@@ -140,13 +142,32 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 			throw new InvalidCredentialsException();
 		}
 
-		await this.usersService.createOne({
+		const userPayload = {
 			provider: this.config.provider,
 			email: email,
 			external_identifier: identifier,
 			role: this.config.defaultRoleId,
 			auth_data: tokenSet.refresh_token && JSON.stringify({ refreshToken: tokenSet.refresh_token }),
-		});
+		};
+
+		// Run hook so the end user has the chance to augment the
+		// user that is about to be created
+		const updatedUserPayload = await emitter.emitFilter(
+			`auth.register`,
+			userPayload,
+			{
+				identifier,
+				provider: this.config.provider,
+				accessToken: tokenSet.access_token,
```

Comment:
We should pass userInfo into the emitter ino order to avoid fetch the userInfo twice.

Reply:
@liho00 I see your point, it could indeed solves some use cases.

@aidenfoxx @rijkvanzanten Does it makes sense to add a `userInfo` meta data prop that would contains the whole userInfo object just retrieved ? We would have no idea what is inside :thinking:

Reply:
I think it's fine to pass it along as long as we make it clear the userInfo can differ between providers.

Reply:
@gkielwasser 
for example, the userInfo included avatart image url, hence we able to pass to the user payload from the userInfo without calling userInfo twice.

Reply:
@gkielwasser nice, @aidenfoxx  @rijkvanzanten should be good to merge.